app/models/piece.rb

```diff
@@ -73,4 +74,24 @@ def move_to!(new_x, new_y)
       self.update(x_position: new_x, y_position: new_y)
     end
   end
+=======
+  def moving_on_board?(x, y)
+    x.between?(0, 7) && y.between?(0, 7)
+  end
+
+  def valid_move?
+    if !(self.moving_on_board?(x, y))
+      return false
+    elsif self.is_obstructed?(x, y) && @pieces_in_the_way.count > 1
+      return false
+    elsif self.is_obstructed?(x, y) && @pieces_in_the_way.first.color == self.color
```

Comment:
I am going to assume this fails since we can't call `.color` on our pieces since color does not exist in our schema. Or did I miss it being added to a helper file somewhere.

Reply:
True, I forget that these words mean nothing to the computer haha